packages/ipfs-core/test/name.spec.js

```diff
@@ -50,6 +50,26 @@ describe('name', function () {
       expect(republisher._republishEntries.calledTwice).to.equal(true)
     })
 
+    it('should not republish self key twice', async function () {
+      const mockKeychain = {
+        listKeys: () => Promise.resolve([{ name: 'self' }])
+      }
+      republisher = new IpnsRepublisher(sinon.stub(), sinon.stub(), sinon.stub(), mockKeychain, {
+        initialBroadcastInterval: 500,
+        broadcastInterval: 1000,
+        pass: 'pass'
+      })
+      republisher._republishEntry = sinon.stub()
+
+      await republisher.start()
+
+      expect(republisher._republishEntry.calledOnce).to.equal(false)
+
+      // Initial republish should happen after ~500ms
+      await delay(750)
```

Comment:
Can we configure a smaller delay to make the tests faster?

Reply:
changed to 200 with initial broadcast of 100

Reply:
also decreased the other test timers